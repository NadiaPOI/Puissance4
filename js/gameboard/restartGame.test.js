beforeEach(() => {
  document.body.innerHTML = `<table id="gameboard></table>
    <section id="displayWinner">
    <p>Great !! Yellow player win !</p>
    <button id="restart">Start again</button>
    </section>
    <aside>
      <form>
        <input type="text" name="row" id="row" value="6">
        <input type="text" name="col" id="col" value="7">
        <input type="submit" value="Generate gameboard">
      </form>
    </aside>
    `;
});

describe('restartGame', () => {
  it('should display a form to submit a board when you click to the button start again', () => {
    let btn = document.getElementById('restart');
    btn.click();

    expect(document.querySelector('aside')).toMatchSnapshot();
  });
});
