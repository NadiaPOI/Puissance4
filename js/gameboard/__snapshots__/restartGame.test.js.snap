// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restartGame should display a form to submit a board when you click to the button start again 1`] = `
<aside>
  
      
  <form />
  
        
  <input
    id="row"
    name="row"
    type="text"
    value="6"
  />
  
        
  <input
    id="col"
    name="col"
    type="text"
    value="7"
  />
  
        
  <input
    type="submit"
    value="Generate gameboard"
  />
  
      
    
</aside>
`;
